aten/src/THC/generic/THCTensorIndex.cu

```diff
@@ -240,18 +258,35 @@ void THCTensor_(indexAdd)(THCState *state, THCTensor *dst, int dim, THCudaLongTe
 
   int mpc = THCState_getCurrentDeviceProperties(state)->multiProcessorCount;
 
-#define SMALL_INDEX(TENSOR_TYPE, TYPE, DST_DIM, SRC_DIM, IDX_DIM) \
-  indexAddSmallIndex<TENSOR_TYPE, TYPE, DST_DIM, SRC_DIM, IDX_DIM> \
-    <<<smallIndexGrid, smallIndexBlock, 0, stream>>>(   \
-      dstInfo, srcInfo, indicesInfo,                    \
-      dstAddDim, srcAddDim, sliceSize, dstAddDimSize);
-
-#define LARGE_INDEX(TENSOR_TYPE, TYPE, DST_DIM, SRC_DIM, IDX_DIM) \
-  indexAddLargeIndex<TENSOR_TYPE, TYPE, DST_DIM, SRC_DIM, IDX_DIM> \
-    <<<largeIndexGrid, largeIndexBlock, 0, stream>>>(   \
-      dstInfo, srcInfo, indicesInfo,                    \
-      dstAddDim, srcAddDim, sliceSize, dstAddDimSize);
-
+#if defined(__HIP_PLATFORM_HCC__)
+  #define SMALL_INDEX(TENSOR_TYPE, TYPE, DST_DIM, SRC_DIM, IDX_DIM) \
+    hipLaunchKernelGGL(                                             \
+      (indexAddSmallIndex<TENSOR_TYPE, TYPE, DST_DIM, SRC_DIM, IDX_DIM>), \
+        smallIndexGrid, smallIndexBlock, 0, stream,                 \
+        make_magic_wrapper(dstInfo), make_magic_wrapper(srcInfo),   \
+        make_magic_wrapper(indicesInfo),                            \
+        dstAddDim, srcAddDim, sliceSize, dstAddDimSize);
+  
+  #define LARGE_INDEX(TENSOR_TYPE, TYPE, DST_DIM, SRC_DIM, IDX_DIM) \
+    hipLaunchKernelGGL(                                             \
```

Comment:
why is the hibLaunchKernelGGL needed here? can you give some context, so that we can see if we can work out a solution that doesn't need such a special-case

Reply:
Unfortunately our parsing tool has issues with converting kernel launches inside multiline macros. Therefore, for these cases, it was necessary to include a separate compilation using the HIP-specific preprocessor guards.